src/app/portlets/dot-edit-page/content/dot-edit-content.component.ts

```diff
@@ -202,23 +198,34 @@ export class DotEditContentComponent implements OnInit {
         });
     }
 
-    private getPageMode(page: DotRenderedPage): PageMode {
-        return page.locked && page.canLock ? PageMode.EDIT : PageMode.PREVIEW;
-    }
-
     private contentletEventsHandler(event: any): Function {
         const eventsHandlerMap = {
-            'edit': this.editContentlet.bind(this),
-            'add': this.addContentlet.bind(this),
-            'remove': this.removeContentlet.bind(this),
-            'select': this.closeDialog.bind(this),
-            'cancel': () => {},
-            'save': () => {},
+            edit: this.editContentlet.bind(this),
+            add: this.addContentlet.bind(this),
+            remove: this.removeContentlet.bind(this),
+            select: this.closeDialog.bind(this),
+            cancel: () => {},
+            save: () => {}
         };
 
         return eventsHandlerMap[event];
     }
 
+    private handleSetPageStateFailed(err: ResponseView): void {
+        this.dotHttpErrorManagerService.handle(err).subscribe((res: any) => {
+            if (res.forbidden) {
```

Comment:
This handle the scenario when you are in a page and you have permissions at least to view, but someone remove the view permission, when any request fails (lock, unlock, renderHTML) we redirect to the site browser.